local-cli/runIOS/runIOS.js

```diff
@@ -64,82 +64,122 @@ function runOnDeviceByUdid(udid, scheme, xcodeProject, devices) {
-  const iosDeployInstallArgs = [
-    '--bundle', 'build/Build/Products/Debug-iphoneos/' + scheme + '.app',
-    '--id' , selectedDevice.udid,
-    '--justlaunch'
-  ];
-  console.log(`installing and launching your app on ${selectedDevice.name}...`);
-  var iosDeployOutput = child_process.spawnSync('ios-deploy', iosDeployInstallArgs, {encoding: 'utf8'});
-  if (iosDeployOutput.error) {
-    console.log('');
-    console.log('** INSTALLATION FAILED **');
-    console.log('Make sure you have ios-deploy installed globally.');
-    console.log('(e.g "npm install -g ios-deploy")');
-  } else {
-    console.log('** INSTALLATION SUCCEEDED **');
-  }
+  return buildProject(xcodeProject, selectedDevice.udid, scheme)
+  .then((appName) =>
+  {
+    if (!appName)
+      appName = scheme;
+    const iosDeployInstallArgs = [
+      '--bundle', 'build/Build/Products/Debug-iphoneos/' + appName + '.app',
+      '--id' , selectedDevice.udid,
+      '--justlaunch'
+    ];
+    console.log(`installing and launching your app on ${selectedDevice.name}...`);
+    let iosDeployOutput = child_process.spawnSync('ios-deploy', iosDeployInstallArgs, {encoding: 'utf8'});
+    if (iosDeployOutput.error) {
+      console.log('');
+      console.log('** INSTALLATION FAILED **');
+      console.log('Make sure you have ios-deploy installed globally.');
+      console.log('(e.g "npm install -g ios-deploy")');
+    } else {
+      console.log('** INSTALLATION SUCCEEDED **');
+    }
+  });
 }
 
 function buildProject(xcodeProject, udid, scheme) {
-  const xcodebuildArgs = [
-    xcodeProject.isWorkspace ? '-workspace' : '-project', xcodeProject.name,
-    '-scheme', scheme,
-    '-destination', `id=${udid}`,
-    '-derivedDataPath', 'build',
-  ];
-  console.log(`Building using "xcodebuild ${xcodebuildArgs.join(' ')}"`);
-  child_process.spawnSync('xcodebuild', xcodebuildArgs, {stdio: 'inherit'});
+  return new Promise((resolve,reject) =>
+  {
+     const xcodebuildArgs = [
+      xcodeProject.isWorkspace ? '-workspace' : '-project', xcodeProject.name,
+      '-scheme', scheme,
+      '-destination', `id=${udid}`,
+      '-derivedDataPath', 'build',
+    ];
+    console.log(`Building using "xcodebuild ${xcodebuildArgs.join(' ')}"`);
+    const buildProcess = child_process.spawn('xcodebuild', xcodebuildArgs);
+    let buildOutput = "";
+    buildProcess.stdout.on('data', function(data) {
+      console.log(data.toString());
+      buildOutput += data.toString();
+    });
+    buildProcess.stderr.on('data', function(data) {
+        console.log(data.toString());
+    });
+    buildProcess.on('close', function(code) {
+      //FULL_PRODUCT_NAME is the actual file name of the app, which actually comes from the Product Name in the build config, which does not necessary match a scheme name,  example output line: export FULL_PRODUCT_NAME="Super App Dev.app"
+      let productNameMatch = /export FULL_PRODUCT_NAME="*(.+).app/.exec(buildOutput);
```

Comment:
Since there's just going to be 1 or 0 `"` in your regex, use `"?` instead of `"*"`